api/pkg/transformer/pipeline/handler.go

```diff
@@ -0,0 +1,52 @@
+package pipeline
+
+import (
+	"context"
+	"encoding/json"
+
+	"github.com/gojek/merlin/pkg/transformer/types"
+)
+
+type Handler struct {
+	compiledPipeline *CompiledPipeline
+}
+
+const PipelineEnvironmentContext = "merlin-transfomer-environment"
+
+func NewHandler(compiledPipeline *CompiledPipeline) *Handler {
+	return &Handler{compiledPipeline: compiledPipeline}
+}
+
+func (h *Handler) Preprocess(ctx context.Context, rawRequest []byte, rawRequestHeaders map[string]string) ([]byte, error) {
+	env := getEnvironment(ctx)
```

Comment:
I passed the environment via context is because `Preprocess` and `Postprocess` must share environment. Another alternative is to modify directly the `Server`.